Packs/Slack/Integrations/SlackV3/SlackV3.py

```diff
@@ -118,6 +121,130 @@ def get_current_utc_time() -> datetime:
+    else:
+        return {}
+
+
+def get_user_by_email(user_to_search: str) -> dict:
+    """
+    Searches for a user when given the user's email.
+
+    Args:
+        user_to_search: the user's email we are searching for.
+    Returns:
+        Formatted user results if a user is found.
+    """
+    if not re.match(emailRegex, user_to_search):
+        raise ValueError('must provide user email')
+
+    _body = {
+        'email': user_to_search
+    }
+    response = send_slack_request_sync(CLIENT, 'users.lookupByEmail', http_verb='GET', body=_body)
+    user = response.get('user', {})
+
+    if not user:
+        demisto.results({
+            'Type': WARNING_ENTRY_TYPE,
+            'Contents': f'User {user_to_search} not found in Slack',
+            'ContentsFormat': formats['text']
+        })
+        return {}
+    else:
+        return format_user_results(user)
+
+
+def add_user_to_context(user: dict, integration_context=None):
+    """
+    In some cases, we want to save user data to the integration context.
+
+    Args:
+        integration_context: The current context for the instance of the Slack which executed the command.
+        user: The context safe version of a user's data to insert into the context.
+    """
+    if integration_context.get('users'):
+        users = json.loads(integration_context['users'])
+        users.append(user)
+    else:
+        users = [user]
+    set_to_integration_context_with_retries({'users': users}, OBJECTS_TO_KEYS, SYNC_CONTEXT)
+
+
+def paginated_search_for_user(user_to_search: str):
+    """
+    When a user cannot be found via the context or users.lookupByEmail, ONLY if SAFE_MODE is disabled, will we attempt
+    to paginate through the Slack users.list results. Please note, for large workspaces, this action can exceed the
+    timeout limit for the command which triggered it.
+
+    Args:
+        user_to_search: String representation of the user we are looking for. Can be user_name, display_name, real_name
+    Returns:
+        If a user is found, will return a context safe version of the user found. Else, will return an empty dict.
+    """
+    demisto.debug(f"Attempting to fetch data for {user_to_search} from Slack API.")
+    body = {
+        'limit': PAGINATED_COUNT
+    }
+
+    response = send_slack_request_sync(CLIENT, 'users.list', http_verb='GET', body=body)
+    while True:
+        workspace_users = response['members'] if response and response.get('members',
+                                                                           []) else []
+        cursor = response.get('response_metadata', {}).get('next_cursor')
+        user = return_user_filter(user_to_search, workspace_users)
+        if user:
+            break
+        if not cursor:
+            break
+        body = body.copy()  # strictly for unit testing purposes
+        body.update({'cursor': cursor})
+        response = send_slack_request_sync(CLIENT, 'users.list', http_verb='GET', body=body)
```

Comment:
this is without retries? let's talk about this.

Reply:
It's embedded in the `send_slack_request_sync` function.